SingularityUI/app/components/common/modalButtons/RunNowModal.jsx

```diff
@@ -35,22 +39,34 @@ class RunNowModal extends Component {
   }
 
   handleRunNow(data) {
+    localStorage.setItem(LOCAL_STORAGE_AFTER_TRIGGER_VALUE, data.afterTrigger);
+    const runId = uuid.v4();
+    data.runId = runId;
+    if (data.afterTrigger === RunNowModal.AFTER_TRIGGER.TAIL.value) localStorage.setItem(LOCAL_STORAGE_TAIL_AFTER_TRIGGER_FILENAME, data.fileToTail);
     this.props.runNow(data).then((response) => {
       if (response.error) {
         Messenger().post({
           message: '<p>This request cannot be run now. This is likely because it is already running.</p>',
           type: 'error'
         });
       } else if (_.contains([RunNowModal.AFTER_TRIGGER.SANDBOX.value, RunNowModal.AFTER_TRIGGER.TAIL.value], data.afterTrigger)) {
+        const requestId = Utils.maybe(response, ['data', 'request', 'id']);
```

Comment:
The poller tries to find the task, but never will because it's requesting `/api/history/request/undefined/run/${runId}`.
It might make sense to not even try to poll if that's undefined.